backend/entityservice/serialization.py

```diff
@@ -54,31 +55,41 @@ def binary_pack_filters(filters, encoding_size):
     """Efficient packing of bloomfilters.
 
     :param filters:
-        An iterable of bytes as produced by deserialize_bytes.
+        An iterable of tuples, with
+            - first element is the entity ID as an unsigned int
+            - second element is 'encoding_size' number of bytes as produced by deserialize_bytes.
 
     :return:
         An iterable of bytes.
     """
     bit_packing_struct = binary_format(encoding_size)
 
     for hash_bytes in filters:
-        yield bit_packing_struct.pack(hash_bytes)
+        yield bit_packing_struct.pack(*hash_bytes)
 
 
 def binary_unpack_one(data, bit_packing_struct):
-    clk_bytes, = bit_packing_struct.unpack(data)
-    return clk_bytes
+    entity_id, clk_bytes, = bit_packing_struct.unpack(data)
+    return entity_id, clk_bytes
 
 
-def binary_unpack_filters(streamable_data, max_bytes=None, encoding_size=None):
+def binary_unpack_filters(data_iterable, max_bytes=None, encoding_size=None):
+    """
+    Unpack filters that were packed with the 'binary_pack_filters' method.
+
+    :param data_iterable: an iterable of binary packed filters.
+    :param max_bytes: if present, only read up to 'max_bytes' bytes.
+    :param encoding_size: the encoding size of one filter, excluding the entity ID info
```

Comment:
Since we bounce between bytes and bits I'd try be explicit about the size.

Reply:
good point